### PR TITLE
MBS-11638 / MBS-11639 / MBS-11640: Fixes for the work attribute form

### PR DESCRIPTION
**Pull request by reosarevok**
### Fix MBS-11638 / MBS-11639 / MBS-11640

See commit messages for details

Tested manually by creating the relevant situations on local data.

### CODE REVIEW

File: root/static/scripts/work.js [hunk 1]
```diff
@@ -121,36 +121,46 @@ class WorkAttribute {
...
-      if (!typeID || this.allowsFreeText()) {
+      if (this.allowsFreeText(typeID)) {
         return [];
       }
       return this.parent.allowedValuesByTypeID[typeID] ?? [];
     });
 
+    this.typeID.subscribe((previousTypeID => {
+      this.previousTypeID = previousTypeID;
+    }), this, 'beforeChange');
```
Comment (reosarevok) [inline root/static/scripts/work.js:140]: This causes an issue with our flow typing for subscribe - the typing needs to be updated to support these extra arguments, AFAICT. Not quite sure how to do so.

**Review by mwiencek — CHANGES_REQUESTED** [review]
Works well. I was thinking it'd make sense to prevent submission and show an error with JS, but that's kind of annoying with Knockout and this probably isn't a super common error, so it could wait.

File: lib/MusicBrainz/Server/Form/Work.pm [hunk 2]
```diff
@@ -145,11 +146,23 @@ after 'validate' => sub {
             next;
         }
 
+        my @used_attribute_values = @{ $used_attributes{$attribute_type} // [] };
```
Comment (mwiencek) [inline lib/MusicBrainz/Server/Form/Work.pm:149]: Try `//= []`, then you can remove the line `$used_attributes{$attribute_type} = \@used_attribute_values;` below.

**Review by mwiencek — APPROVED** [review]

**Review by yvanzo — COMMENTED** [review]
Commit about MBS-11638 works for me. I did not find how to reproduce MBS-11639. There is no mention of MBS-11640 in commits. (Reran CI test to make it pass.)

**Comment by reosarevok** [comment]
@yvanzo: for MBS-11639 you can just pick "Key", select whatever value, then change the selection to one of the free text attributes (any of the identifiers). You should see the value ID for the previously selected key as the free text value (at least on Chrome). I seemingly squashed MBS-11640 into the MBS-11639 commit by mistake, will split.

**Pull request merged**